tests/unit/reports/single_table/_properties/test_synthesis.py

```diff
@@ -0,0 +1,94 @@
+from unittest.mock import Mock, call, patch
+
+import pandas as pd
+
+from sdmetrics.reports.single_table._properties.synthesis import Synthesis
+
+
+class TestSynthesis:
+
+    @patch('sdmetrics.reports.single_table._properties.synthesis.'
+           'NewRowSynthesis.compute_breakdown')
+    def test__generate_details(self, newrowsynthesis_mock):
+        """Test the ``_generate_details`` method."""
+        # Setup
+        real_data = pd.DataFrame({
+            'col1': [1, 2, 3, 4],
+            'col2': [False, True, True, False],
+            'col3': ['a', 'b', 'c', 'd'],
+            'col4': pd.to_datetime(['2020-01-01', '2020-01-02', '2020-01-03', '2020-01-04'])
+        })
+        synthetic_data = pd.DataFrame({
+            'col1': [1, 7, 3, 4],
+            'col2': [False, True, True, False],
+            'col3': ['a', 'b', 'c', 'd'],
+            'col4': pd.to_datetime(['2020-01-01', '2020-01-02', '2020-01-03', '2020-01-04'])
+        })
+        metadata = {
+            'columns': {
+                'col1': {'sdtype': 'numerical'},
+                'col2': {'sdtype': 'boolean'},
+                'col3': {'sdtype': 'categorical'},
+                'col4': {'sdtype': 'datetime'}
+            }
+        }
+
+        newrowsynthesis_mock.return_value = {
+            'score': 0.25,
+            'num_matched_rows': 3,
+            'num_new_rows': 1,
+        }
+        # Run
```

Comment:
New line

Reply:
Yes, done in [b00d7ca](https://github.com/sdv-dev/SDMetrics/pull/398/commits/b00d7cafd54bf24f3b76c78763bddbd65d5b9408)